src/wp-admin/includes/plugin.php

```diff
@@ -1203,26 +1203,39 @@ function validate_plugin_requirements( $plugin ) {
 	}
 
 	if ( WP_Plugin_Dependencies::has_unmet_dependencies( $plugin ) ) {
-		$dependencies       = WP_Plugin_Dependencies::get_dependencies( $plugin );
-		$unmet_dependencies = array();
+		$dependency_names       = WP_Plugin_Dependencies::get_dependency_names( $plugin );
+		$unmet_dependencies     = array();
+		$unmet_dependency_names = array();
 
-		foreach ( $dependencies as $dependency ) {
+		foreach ( $dependency_names as $dependency => $dependency_name ) {
 			$dependency_file = WP_Plugin_Dependencies::get_dependency_filepath( $dependency );
 
 			if ( false === $dependency_file ) {
-				$unmet_dependencies['not_installed'][] = $dependency;
+				$unmet_dependencies['not_installed'][ $dependency ] = $dependency_name;
+				$unmet_dependency_names[]                           = $dependency_name;
 			} elseif ( is_plugin_inactive( $dependency_file ) ) {
-				$unmet_dependencies['inactive'][] = $dependency;
+				$unmet_dependencies['inactive'][ $dependency ] = $dependency_name;
+				$unmet_dependency_names[]                      = $dependency_name;
 			}
 		}
 
+		$plugins_page  = is_multisite() ? network_admin_url( 'plugins' ) : admin_url( 'plugins' );
```

Comment:
Updated in [7d3b8c9](https://github.com/WordPress/wordpress-develop/pull/6074/commits/7d3b8c9c8c2215dd507ae71a2257af9df61762cb). How's that?

Reply:
Also, I changed the parameter to the `(network_)admin_url()` calls to `'plugins.php'`, as that's the correct location. Noticed the mistake while I was adjusting the messaging.

Reply:
That looks good. The only thing I'm not sure of is if concatenating the strings together is good for l18n. I think it may be better to include the full string for contextual purposes.

Reply:
Yeah I was unsure about that. Using the full string means the `_n()` blocks would be repeated quite a bit so if we can avoid it without causing difficulty, that would be great. "Manage plugins" and "Please contact your network administrator" seem like fine standalone strings. The concatenated space bothers me though.

 @swissspidy Do you have any thoughts on this?

Reply:
Not uncommon for core to concatenate the strings in cases like this where it's two separate sentences like here.

So fine by me 👍